test/Npgsql.Tests/WriteBufferTests.cs

```diff
@@ -52,6 +52,35 @@ public void ChunkedByteArrayEncodingFits()
             Assert.That(completed, Is.False);
         }
 
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/3733")]
+        public void ChunkedStringEncodingFitsWithSurrogates()
+        {
+            WriteBuffer.WriteBytes(new byte[WriteBuffer.Size - 1], 0, WriteBuffer.Size - 1);
```

Comment:
nit: can omit the last two parameters (i.e. use the ReadOnlySpan overload):

```suggestion
            WriteBuffer.WriteBytes(new byte[WriteBuffer.Size - 1]);
```